pkgs/applications/misc/mario/default.nix

```diff
@@ -0,0 +1,54 @@
+{ lib, poetry2nix, fetchFromGitHub, python3Packages }:
+
+poetry2nix.mkPoetryApplication rec {
+  src = python3Packages.fetchPypi {
+    pname = "mario";
+    version = "v0.0.155";
+    sha256 = "0ks95wchym32bv6h8cdqzzx67z5f46gxhc1k704w58i74vygm8p1";
+  };
```

Comment:
Wait no, false diagnosis. The problem is that poetry2nix can't use `builtins.fromTOML` without IFD if the file it needs to read isn't built from a derivation, but this is exactly what's done here.

So instead what you need to do is copy the `pyproject.toml` and `poetry.lock` files into the `mario` directory, then assigning `projectDir = ./.`, so those files are used instead. Then no IFD is needed. This is also how poetry itself is packaged in nixpkgs, see https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/tools/poetry2nix/poetry2nix/pkgs/poetry for an example.